micro-rdk/src/common/data_store.rs

```diff
@@ -253,7 +253,14 @@ impl DataStore for DefaultDataStore {
         }
         let encode_len = message.encoded_len();
         let total_encode_len = length_delimiter_len(encode_len) + encode_len;
-
+        let buffer_capacity = buffer.capacity();
```

Comment:
The first time I read this code, I thought I was looking at a bug, because why would you be comparing the total encoded length to the _capacity_ of the buffer, rather than the available length. But I realized after looking a little more closely that it is in fact correct because after the loop below starting on line 264, `buffer.vacant_len` will eventually reach `buffer.capacity` as messages are dropped. So checking against `capacity` here is correct, but I think a comment is warranted.

Reply:
+1 it can be difficult to decipher